File: dsa/cp/lC/21. Merge Two Sorted Lists.java
import java.lang.classfile.components.ClassPrinter.ListNode;
import java.util.LinkedList;
import java.util.stream.Gatherer.Integrator;

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * Input: list1 = [1,2,4], list2 = [1,3,4]
   Output: [1,1,2,3,4,4]
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
      LinkedList<Integer> ll=new LinkedList<>();
      for(int i=0;i<list1.size();i++){
        ll.add(i,list1.get(i))
      }
    }
}

File: LeetcodeDaily/X/21.MergeTwoSortedLists.py
# Definition for singly-linked list.
class listNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def mergeTwolists(self, list1: Optional[listNode], list2: Optional[listNode]) -> Optional[listNode]:


        dummy=listNode()

        curr=dummy

        while list1 and list2:
            # print(curr)
            if list1.val < list2.val:
                curr.next=list1
                list1=list1.next
            else:
                curr.next=list2
                list2=list2.next
            curr=curr.next

        if list1:
            curr.next=list1
        if  list2:
            curr.next=list2
        print(dummy)
        return dummy.next
 
